gamemode/sv_bannedmodels.lua

```diff
@@ -1,102 +1,94 @@
-GM.BannedModels = {}
+-- This file is what controls what models are banned. Models that are banned
+-- cannot be chosen as a disguise.
+
+
+GM.BannedModels = {} -- This is used as a hash table where the key is the model string and the value is true.
+
+
+util.AddNetworkString("ph_bannedmodels_getall")
+util.AddNetworkString("ph_bannedmodels_add")
+util.AddNetworkString("ph_bannedmodels_remove")
 
-util.AddNetworkString("ph_bannedmodels")
 
 function GM:IsModelBanned(model)
-	return table.HasValue(self.BannedModels, model)
+	return self.BannedModels[model] == true
 end
 
+
 function GM:AddBannedModel(model)
-	table.insert(self.BannedModels, model)
+	self.BannedModels[model] = true
```

Comment:
I know you just removed the check here (and below), but given it does save a file maybe it's worth doing the check.

Reply:
Re-added the checks.